src/main/java/core/basesyntax/dao/impl/FruitStorageDaoImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitStorageDaoImpl implements FruitStorageDao {
+    @Override
+    public void addFruit(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitStorage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
+    }
+
+    @Override
+    public void increaseQuantity(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitStorage.merge(fruitTransaction.getFruit(),
+                fruitTransaction.getQuantity(), Integer::sum);
+    }
+
+    @Override
+    public void decreaseQuantity(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitStorage.merge(fruitTransaction.getFruit(),
+                -fruitTransaction.getQuantity(), Integer::sum);
```

Comment:
When decreasing the quantity of fruit, you should check if the resulting quantity is not negative. According to the checklist, the balance should be positive. If the balance after the decrease operation is negative, you should throw a `RuntimeException`.